src/equations/linear_scalar_advection_1d.jl

```diff
@@ -91,43 +91,6 @@ function initial_condition_sin(x, t, equation::LinearScalarAdvectionEquation1D)
 end
 
 
-"""
-    initial_condition_linear_x(x, t, equations::LinearScalarAdvectionEquation1D)
-
-A linear function of `x[1]` used together with
-[`boundary_condition_linear_x`](@ref).
-"""
-function initial_condition_linear_x(x, t, equation::LinearScalarAdvectionEquation1D)
```

Comment:
I am not a fan of removing these ICs and BCs (also for the 2D case). I know they are simple and stupid, but that's why they are also extremely useful for trying out new numerical methods, features etc. For example, if a student were to implement a new solver type and they have errors, I'd tell them for debugging, start with linear advection and then make sure that - in this order - the following things work:
* IC constant
* IC linear in x
* IC linear in y
* IC with sin
etc. Thus while they are not really needed for running simulations, I'd classify them as "debugging tools" and thus argue against their removal.

Reply:
In that case, they are an idea candidate for something like `Trixi/dev/debugging_linear_advection_1d.jl`, aren't they? Do we really need them in `Trixi/src` itself? (I removed them in one single commit to be able to `git revert` that one later, but we might still discuss it here)

Reply:
I can also re-enable these tests here and we can discuss this somewhere else to move forward

Reply:
I like this suggestion :+1: